server/pkg/key/key.go

```diff
@@ -32,3 +33,9 @@ func (k Key) IsValid() bool {
 func (k Key) String() string {
 	return k.key
 }
+
+func (k Key) Clone() Key {
+	return Key{
+		key: strings.Clone(k.key),
```

Comment:
```suggestion
		key: k.key,
```

string is not a reference, so it does not need to be cloned